### ISSUE TITLE
hosts: {{ host_name }} cause errors

### ISSUE THREAD

**Issue by ReinerNippes** [ansible/ansible-lint#2872]
# Issue Type

- Bug report

# Ansible and Ansible Lint details

```
ansible --version
  ansible [core 2.13.3]
ansible-lint --version
  ansible-lint 6.0.2 using ansible 2.13.3
```

- ansible installation method: OS package
- ansible-lint installation method: pip, OS package

# Desired Behavior

ansible-lint should process the following playbook
```
- name: just an example
  hosts: "{{ host_name }}"

  tasks:
    - name: just a task
      debug: msg="Hello world."
```

# Actual Behavior (Bug report only)

Error message:

```
$ ansible-lint debug_lint.yml 
WARNING  Listing 1 violation(s) that are fatal
syntax-check: The field 'hosts' has an invalid value, which includes an undefined variable. The error was: 'host_name' is undefined
debug_lint.yml:1:3 ERROR! The field 'hosts' has an invalid value, which includes an undefined variable. The error was: 'host_name' is undefined

The error appears to be in '/home/.../debug_lint.yml': line 1, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


- name: just an example
  ^ here

Finished with 1 failure(s), 0 warning(s) on 1 files.
```

Applying -e host_name=localhost doesn't word since --extra-vars isn't a valid option for ansible-lint.

```
ansible-lint debug_lint.yml --extra-vars host_name=localhost
usage: ansible-lint [-h] [-L] [-f {rich,plain,rst,json,codeclimate,quiet,pep8}] [-q] [-p] [--progressive] [--project-dir PROJECT_DIR] [-r RULESDIR] [-R] [--write]
                    [--show-relpath] [-t TAGS] [-T] [-v] [-x SKIP_LIST] [-w WARN_LIST] [--enable-list ENABLE_LIST] [--nocolor] [--force-color]
                    [--exclude EXCLUDE_PATHS] [-c CONFIG_FILE] [--offline] [--version]
                    [lintables ...]
ansible-lint: error: unrecognized arguments: --extra-vars host_name=localhost
```

**Comment by richm** [comment]
Does it work if you add `extra_vars` to `.ansible-lint` as in https://ansible-lint.readthedocs.io/configuring/#using-local-configuration-files ?
```yaml
extra_vars:
  host_name: localhost
```
?

**Comment by MarkusTeufelberger** [comment]
As a workaround, I fixed similar constructs in my playbooks like this:

```
- name: just an example
  hosts: "{{ host_name | default('localhost') }}"

  tasks:
    - name: just a task
      debug: msg="Hello world."
```

This still breaks when trying to run import_playbook on such a file even when supplying `host_name` in `vars`, but at least it works for directly linting it.

**Comment by ssbarnea** [comment]
I guest that one is just a RTFD one as https://ansible-lint.readthedocs.io/rules/syntax-check/#correct-code already contains exactly the solution for that problem.
